src/datachain/lib/signal_schema.py

```diff
@@ -117,40 +146,119 @@ def from_column_types(col_types: dict[str, Any]) -> "SignalSchema":
                 )
         return SignalSchema(signals)
 
-    def serialize(self) -> dict[str, str]:
-        signals = {}
+    @staticmethod
+    def serialize_custom_model_fields(fr: type, custom_types: dict[str, Any]) -> str:
+        """This serializes any custom type information to the provided custom_types
+        dict, and returns the name of the type provided."""
+        orig = get_origin(fr)
+        if orig in (Literal, LiteralEx):
+            # Literal has no __name__ in Python 3.9
+            name = "Literal"
+        else:
+            name = str(fr.__name__)
+        if (
+            not ModelStore.is_pydantic(fr)
+            or not issubclass(fr, BaseModel)
```

Comment:
Isn't this already covered by `not ModelStore.is_pydantic(fr)`?

Reply:
Yes, but the type checker kept throwing errors, so I added this in addition. I changed this to just include the checks from `is_pydantic` instead so there are no duplicate calls.